cli/internal/cmd/spinner.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright (c) Edgeless Systems GmbH
+
+SPDX-License-Identifier: AGPL-3.0-only
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/spf13/cobra"
+)
+
+var (
+	spinnerStates = []string{"⣾", "⣽", "⣻", "⢿", "⡿", "⣟", "⣯", "⣷"}
+	dotsStates    = []string{".", "..", "..."}
+)
+
+type Spinner struct {
+	out      *cobra.Command
+	text     string
+	showDots bool
+	delay    time.Duration
+	wg       *sync.WaitGroup
+	stop     int32
+}
+
+func NewSpinner(c *cobra.Command, text string, showDots bool) *Spinner {
+	return &Spinner{
+		out:      c,
+		text:     text,
+		showDots: showDots,
+		wg:       &sync.WaitGroup{},
+		delay:    100 * time.Millisecond,
+		stop:     0,
+	}
+}
+
+func (s *Spinner) Start() {
+	s.wg.Add(1)
+	go func() {
+		defer s.wg.Done()
+
+	out:
+		for {
+			for i := 0; i < len(spinnerStates); i++ {
+				if atomic.LoadInt32(&s.stop) == 0 {
+					dotsState := ""
+					if s.showDots {
+						dotsState = dotsStates[i%len(dotsStates)]
+					}
+					state := fmt.Sprintf("\r%s %s%s", spinnerStates[i], s.text, dotsState)
+					s.out.Print(state)
+					time.Sleep(s.delay)
+				} else {
+					break out
+				}
+			}
+		}
+
+		dotsState := ""
+		if s.showDots {
+			dotsState = dotsStates[len(dotsStates)-1]
+		}
+		finalState := fmt.Sprintf("\r%s%s", s.text, dotsState)
```

Comment:
There is a tiny bug in this line, as the previous text isn't fully overwritten in case that the dots aren't printed. It produces the following output:
```
Initializing cluster r 
Your Constellation cluster was successfully initialized.
```
The `r` from the previous iteration isn't overwritten as the string is now shorter than before. You can fix this by just printing a space character after the text.
```suggestion
		finalState := fmt.Sprintf("\r%s%s ", s.text, dotsState)
```